### PR TITLE
Remove deprecated theme channel

### LINKED ISSUE #1618: Remove deprecated ThemeProvider channel
We have had this deprecated theme channel for a while now, and it should be removed.

It's quite a simple task since nothing should be depending on it anymore, but it does involve some "housekeeping" task of checking whether there's any leftovers, renaming the new `CHANNEL`, and checking libraries that might depend on it still.

https://github.com/styled-components/styled-components/blob/master/src/models/ThemeProvider.js#L80-L88

### PR DESCRIPTION
**Pull request by matedeilo**
Resolve #1618

Hi, This is my first PR. Trying to remove channel deprecated.

### CODE REVIEW

File: src/hoc/withTheme.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import hoistStatics from 'hoist-non-react-statics'
 import {
-  CHANNEL,
   CHANNEL_NEXT,
```
Comment (kitten) [inline src/hoc/withTheme.js]: We can safely rename this to `CHANNEL` now, since the old one is removed

**Comment by kitten** [comment]
@matedeilo Hiya, thanks for picking this up; see the above comment

Also we need to update withTheme as well: https://github.com/styled-components/styled-components/blob/master/src/hoc/withTheme.js

Can you also include a changelog entry please?

**Comment by matedeilo** [comment]
Hi @kitten  yes, no problem.

**Comment by probablyup** [comment]
@matedeilo hi hi, are you able to pick this up to finish it out?

**Comment by alexandernanberg** [comment]
I could pick this up if @matedeilo is unable. Do I create a new PR in that case or do I layer the commits on this PR somehow?

**Comment by matedeilo** [comment]
@alexandernanberg feel free to finish this.

**Comment by alexandernanberg** [comment]
Cool! @probablyup How do you think will be the best way to accomplish this? The absolute easiest way would to just create a new PR without the existing commits but then they would obviously be lost. Otherwise I guess I need write access on the fork?

**Comment by mxstbr** [comment]
> Otherwise I guess I need write access on the fork?

Pull these commits, then do a `git checkout` to a new branch—which'll preserve the existing commits, but you can then push them to your own fork!

**Comment by probablyup** [comment]
Now that we’re edging toward 4.0 work this should just be superseded by
switching to the new context api I think

**Comment by alexandernanberg** [comment]
Okay, yea seems smarter to jump on the new Context API directly in that case. I've begun migrating SC to that on a local branch but I've hit some problems with Enzyme and `StyledComponent.js`, is it worth opening a PR even though I'm not sure I'll be able to tackle this alone?

**Comment by probablyup** [comment]
That’s up to you! If you’ve started the work we definitely won’t say no to
getting the ball rolling. The final product might look a bit different. I’m
going to make a “develop” branch for 4.0 sometime today

**Comment by probablyup** [comment]
Superceded by the new context API work merged into the develop (4.0) branch

**Pull request closed without merging**